tests/integrations/raftstore/test_transfer_leader.rs

```diff
@@ -108,6 +109,46 @@ fn test_pd_transfer_leader<T: Simulator>(cluster: &mut Cluster<T>) {
         assert!(!resp.get_header().has_error(), "{:?}", resp);
         assert_eq!(resp.get_responses()[0].get_get().get_value(), b"v");
     }
+
+    // with multi targets
+    for id in 1..4 {
+        // select multi target leaders to transfer
+        // set `peer` 0 to make sure the new leader must come from `peers`
+        pd_client.transfer_leader(
+            region.get_id(),
+            new_peer(0, 0),
+            vec![new_peer(id, id), new_peer((id % 3) + 1, (id % 3) + 1)],
+        );
+
+        for _ in 0..100 {
+            // reset leader and wait transfer successfully.
+            cluster.reset_leader_of_region(1);
+
+            sleep_ms(20);
+
+            if let Some(leader) = cluster.leader_of_region(1) {
+                if leader.get_id() == id || leader.get_id() == (id % 3) + 1 {
```

Comment:
We should also test that tikv will choose the peer with the highest matched index as the next leader, we could do that by adding an `IsolationFilter` to a store then write some data and test that leadership will transfer to another store.

Reply:
@NingLin-P done. PTAL